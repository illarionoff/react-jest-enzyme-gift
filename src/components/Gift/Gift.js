import React, { Component } from "react";

export default class Gift extends Component {
  state = {
    person: "",
    present: ""
  };

  render() {
    return (
      <div>
        <form action="">
          <input
            className="input-person"
            type="text"
            name="person"
            value={this.state.person}
            onChange={event => {
              this.setState({ person: event.target.value });
            }}
          />
          <input
            className="input-present"
            type="text"
            name="present"
            value={this.state.person}
            onChange={event => {
              this.setState({ present: event.target.value });
            }}
          />
        </form>
        <button
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          Remove Gift
        </button>
      </div>
    );
  }
}
