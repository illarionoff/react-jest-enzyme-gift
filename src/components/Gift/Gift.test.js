import React from "react";
import { shallow } from "enzyme";
import Gift from "./Gift";

describe("Gift component", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id }, removeGift: mockRemove };
  const wrapper = shallow(<Gift {...props} />);

  test("should render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should initialize a person and present in 'state'", () => {
    expect(wrapper.state()).toEqual({ person: "", present: "" });
  });

  describe("when typing into input fields", () => {
    const person = "Uncle";
    const present = "Car";
    beforeEach(() => {
      wrapper
        .find(".input-person")
        .simulate("change", { target: { value: person } });
      wrapper
        .find(".input-present")
        .simulate("change", { target: { value: present } });
    });
    test("should update 'person' in 'state'", () => {
      expect(wrapper.state().person).toEqual(person);
    });

    test("should update 'present' in 'state'", () => {
      expect(wrapper.state().present).toEqual(present);
    });
  });
  describe("when clicking 'Remove Gift' button", () => {
    beforeEach(() => {
      wrapper.find(".btn-remove").simulate("click");
    });

    test("calls remove gift callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});
