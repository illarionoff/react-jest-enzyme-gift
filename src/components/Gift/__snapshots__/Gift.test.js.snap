// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift component should render without error 1`] = `
<div>
  <form
    action=""
  >
    <input
      className="input-person"
      name="person"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="input-present"
      name="present"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
  <button
    className="btn-remove"
    onClick={[Function]}
  >
    Remove Gift
  </button>
</div>
`;
